lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp

```diff
@@ -1671,43 +1671,84 @@ DWARFASTParserClang::ParseStructureLikeDIE(const SymbolContext &sc,
     attrs.is_forward_declaration = true;
   }
 
+  DWARFDIE def_die;
+  if (attrs.is_forward_declaration && cu_language == eLanguageTypeObjC) {
```

Comment:
This is pretty much copied from `ParseEnum`

Reply:
You'd also need to check for `eLanguageTypeObjCPlusPlus` here.

Reply:
Would also be helpful to add a comment explaining why this block exists.